### PR TITLE
Adrenals reworked

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Old:
 - Brought you out of any stun immediately, including stam crit.
 - Gave you 10u omnizine which healed you for quite a while.
 - Gave you stun reductions.
 - Gave you silly runspeed.
 - Lasted for 100 seconds.
 
 
New:
 - Gives you a moderate stamina regen.
 - Gives you a moderate run speed increase.
 - Makes you ignore damage slowdown.
 - Gives you stun resistance (same as pump-up).
 - Gives you sleep immunity.
 - Causes you to take toxin and oxy damage.
 - Makes your screen shake.
 - Makes you drop stuff.
 - Lasts 30 seconds. (Nanites 15 seconds)

Changes are made to try to push adrenals as a "get out of combat" implant instead of a combat implant. All in all, you can make a better adrenal in chemistry at the end of the day.
The stimpack has not been touched, it is already much less versatile.

**Edit: Added changes to ling adrenals.**
Follows much the same philosophy, but is inherently stronger than the traitor one. No dropping, less screen shake, less duration (15 seconds) to allow for getting back in it faster. Also kept the insane speed buff, and extended it by 2 seconds. 

## Why It's Good For The Game
It fits better with the direction stuns are going, and should feel a bit less bullshitty.

## Changelog
:cl: Skoglol
balance: Adrenal implant reworked. Now intended to assist in running away from combat instead of helping you win the fight. Nanite adrenals are the same with half the duration.
balance: Changeling adrenaline reworked. Much the same as traitor adrenal implant above, but with less drawback. Speedboost kept, extended from 4 to 6 seconds.
/:cl:

### CODE REVIEW

**Comment by Ratvar** [comment]
But if you **don't** add drop, you'll just use adrenals to rush your enemy with stunbaton/stunprod/esword, like you did before. Dropping items is needed if you want any speed-related items to be something else then attack chems.
"Since we already have a getaway chem we don't need another." - you forgot to say why

**Comment by Ratvar** [comment]
...why did you thumbs up the PR, then, if the whole point is to "assist in running away from combat instead of helping you win the fight."?

**Comment by kriskog** [comment]
Nowhere is it stipulated that no effort tc items have to be better than stuff available to crew  through effort. Please dispense of this notion.

The drop stays. This is supposed to be a buy for getting out of fights, not staying in them. If you want that kind of thing, the new adrenals aren't for you.

30 seconds not counting whatever stam crit you may have already accrued when popping it, and the speed gain really isnt that big. Keep in mind the bad shit sticks around for the full 30 seconds too, which makes it harder to use this for quick hit and runs.

**Comment by kriskog** [comment]
> 
> 
> > But if you **don't** add drop, you'll just use adrenals to rush your enemy with stunbaton/stunprod/esword, like you did before. Dropping items is needed if you want any speed-related items to be something else then attack chems.
> > "Since we already have a getaway chem we don't need another." - you forgot to say why
> 
> Because it's FINE to have this as an attack chem, especially for nuke ops, who should be ENGAGING sec in combat, not RUNNING FROM THEM

Take the stimpack then.

**Comment by imsxz** [comment]
yes

**Comment by dr-bee** [comment]
I would remove the damage TBH. If you want a damage based downside have it mask your health status like miners salve.

The drop and screen shake should be enough to dis-incentivize using new adrenals for combat rushing.

Nuke ops should keep the old version however.

**Comment by PKPenguin321** [comment]
👍 to these being brought up to code with stamina
👎 to the dropping and changing healing to damage, which seems like a really odd choice. I could see removing the healing but adding damage is ?? why

**Comment by kriskog** [comment]
One activation will run for 15 chem cycles. That means 7.5 toxin damage and some temporary oxyloss. Hardly anything to write home about, though the oxyloss rng can screw you if you are already hurt and/or very unlucky. Could probably tone the prob down a little, I'll sleep on it.

**Comment by PKPenguin321** [comment]
Ok sure but... Why? I'm not asking *how* the damage works, I'm asking why it's there at all

**Comment by kriskog** [comment]
Why shouldn't it be? Mechanically it makes chaining charges worse since you need some time to get that oxyloss down, and I kinda like that.

**Comment by PKPenguin321** [comment]
That's fair

**Comment by necromanceranne** [comment]
Ah lovely I'll just drop this l6 I cannot put in any bag and die since I'm unarmed anyway.

The damage is enough. Let's not spend 8 tc to just fucking die from bad luck and lose our equipment guaranteed. There is never a situation where you are retreating and not fighting. But screen shake makes fighting fucking impossible sometimes anyway and it's for 30 seconds.

I actually believe you should shorten that time up. 10 or 15 seconds so you don't poison yourself completely.

**Comment by Shadowflame909** [comment]
Do what anne says and seriously lower the TC

A massive nerf aint an overhaul. Be honest

**Comment by AdamElTablawy** [comment]
pairing drop chance with screenshake is overkill, screenshake already makes it nearly impossible to fight effectively and the other nerfs make it effective for 'not for in combat use, meant for fleeing situation'

**Comment by AdamElTablawy** [comment]
wait hold on what the fuck why have you added a five second paralyze to being put into stamina crit

**Comment by CPTANT** [comment]
All the movement speed increases of drugs are too strong. 

Nuke ops with adrenals (war or not) is just aids and meth is also completely OP when mixed with some mannitol. Both allow you to yolo rush someone and kill them before they even noticed you entered their view while in turn making you virtually immune to any return fire.

**Comment by kriskog** [comment]
> 
> 
> Ah lovely I'll just drop this l6 I cannot put in any bag and die since I'm unarmed anyway.

You only drop shit from active hand, this is true for ephedrine too. 

> 
> 
> wait hold on what the fuck why have you added a five second paralyze to being put into stamina crit

That is explained in the OP.

**Comment by XDTM** [comment]
You might want to mirror these changes into nanite adrenals

**Comment by AdamElTablawy** [comment]
fuck it 

after an hour long argument where you did nothing but argue in bad faith i'll just dump a small wall of text here

your paralyze change makes stamina crit impossible to escape as long as someone is attacking you with any form of stamina damage even if you're on an anti-stamina and anti-stun chemical 

take, for example, methamphetamine and aransep

noslowdown and decent stamina regen

you heal out of stamina crit immediately on mob life, which you want to avoid

however, you are still stunned by being paralyzed for 5 seconds, which means you are still stuck on the floor

meth reduces this to two seconds

you are still on the floor for two seconds before you stand up, still slow from the stamina damage

now, here's the issue-you're just going to get disablered, stunbatonned, etc while on the floor for those two seconds, putting you back in stamina critical and thus stunning you for two seconds again
this process repeats forever as long as someone is stunning you, thus meaning that for all the effort of getting antistun and antistamina chemicals, you're still chainstunned anyway unable to move. stunned. on the floor. you can't move or escape despite being on antistuns.

i said that about seven times and you kept insisting that nothing had changed and that you couldn't escape from stam crit with meth anyway and that taking stamina reducing chemicals shouldn't make you immune to sec weapons but you've missed the crucial point that you **can't escape stamina crit as long as someone is trying to keep you in it.**

meth won't make a difference in 'reducing the stun timer' of the paralyze-you're still paralyzed for two seconds and can thus be shot to put you in stam crit again resetting the cycle.

meth + aransep won't matter because of the above.

meth + aransep + morphine + coffee won't matter because despite theoretically if you stood up from stamina crit you'd be able to run away at meth speed without the damage slowdown, you won't be able to stand up from the stamina crit because you'll still be fucking stunned and thus will be kept in stamina crit by anyone who even marginally knows what they're doing.

it makes combat drugs like meth and crank and aransep useless because they no longer fulfill their intended purpose, which is letting you escape from being stunned, since you'll still be on the fucking floor getting shot.

if you intend to make it so that being put into stamina crit is intentionally inescapable and want combat drugs to help you escape the fight or avoid being put into stamina crit, but not get you out of it, do that in this pr. otherwise what you're doing is going to nerf every single antistun/antistam chem there is.

**Comment by necromanceranne** [comment]
Or better yet, atomize it into a separate pr.

**Comment by kriskog** [comment]
I already said this a few times, but most anti stun chems will in most situations bring you out of the paralyze on the first chem tick, at the same time as your stam regen brings you out of stam crit. Just repeatedly yelling at me won't change that, regardless if it is on discord or here. 
I also do not agree with you that you should be able to get out of a consistently reapplied stam crit without the chems to do so.

**Comment by AdamElTablawy** [comment]
Most anti stun chems in most situations won't bring you out of the paralyze on the first chem tick. Repeating something that's false won't make it true.

**Comment by kriskog** [comment]
You gotta remember chem ticks run every two server seconds. Which means 50% of the time, you'll be waiting 1 or more second for the chem tick no matter what. So meth (with -4s paralyze) for example, will bring you out of the 5 second paralyze on the first chem tick in 50% of the cases. If we assume you got batoned just before a chem tick, you will be stunned for one second after the tick hits. In both of these cases, you are stunned for 1-2 seconds, depending on how far away from a chem tick you got stunned. This is near enough the same to how it currently works since you will have to wait for the chem tick that it doesn't make the enormous difference you are implying it will. 

Yes, there are some chems that are worse than meth, like crank and synaptizine. They will be worse than the above. If we consider a chem with -2 seconds paralyze, you are still only down for 2-3 seconds because of how the chemticks work out.

I am fine with this.

**Comment by AdamElTablawy** [comment]
you're down for 2-3 seconds, which translates to infinity seconds as long as someone is shooting at you.

**Comment by Shadowflame909** [comment]
Should put all of the major nerfs and downsides in a separate PR.

Feels less like an overhaul and more like an 8 tc item being throttled into uselessness.

File: code/modules/mob/living/carbon/status_procs.dm [hunk 1]
```diff
@@ -14,6 +14,7 @@
 	if(!IsParalyzed())
 		to_chat(src, "<span class='notice'>You're too exhausted to keep going...</span>")
 	stam_paralyzed = TRUE
+	Paralyze(50)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/status_procs.dm]: since this touches things outside of adrenals it should be separate PR tbh.

Comment (kriskog) [reply code/modules/mob/living/carbon/status_procs.dm]: That would require that change be merged before this. I am not okay with making adrenals a sustained 100% anti-stun against sec weaponry, which we then have to fix afterwards.

Comment (PKPenguin321) [reply code/modules/mob/living/carbon/status_procs.dm]: It's fine dude

**Comment by kriskog** [comment]
Fine, #46323 handles the stamcrit changes. I did it differently as well.

I still don't want this merged until the stam crit change goes in, it would make this a super unfair traitor item. I am also dragging ling adrenals into this PR, may as well go for the trifecta.

**Comment by Tlaltecuhtli** [comment]
10 stamina heal/ tick is a joke a disabler which is super spammy does 25 in 1 hit, why make a traitor item which is supposed to counter stamina/stuns dealing bad at its job? it already has limited usage
buff the stamina healing to 30 at least (maybe decrease the stamina health overtime )

**Comment by Shadowflame909** [comment]
What is the Goal of this PR.

This is a very big antag nerf that you want to be merged after a security buff and not before it

Not worth the TC. If escaping with adrenal implants becomes lethal. It just wont be used over cuff breaking implants.

**Comment by kriskog** [comment]
> 
> If escaping with adrenal implants becomes lethal.

It wont be. Just don't chain all three, and you'll be perfectly fine.

**Comment by AdamElTablawy** [comment]
and now you've inexplicably decided to ruin ling adrenals despite lings not having an alternative like traitors do in terms of stimpak

**Comment by ExcessiveUseOfCobblestone** [comment]
The chems aren't meant to fill the same niche.

Previously they were meant to btfo stuns because stuns required a hardcounter.

Stamina does not, so it makes no sense to keep these chems at hardcounter status but now for stamina instead of stuns.

**Comment by Shadowflame909** [comment]
Doesn't even feel much different with 2 second till a stun stunbatons.

**Comment by kriskog** [comment]
If I am being honest, what I have done to ling adrenals scare me a bit. Lings with no damage slowdown in exchange for a moderate screen shake is probably too strong to be honest.

As for stun batons, they don't even hard stun anymore. It's a knockdown after 2 seconds, then a stam crit if you get hit again within 10 seconds.

**Comment by Shadowflame909** [comment]
>Too Strong

How does /tg/ attract so many different opinions to the point of suspected sabotage/trolling.

**Comment by kriskog** [comment]
> 
> 
> > Too Strong
> 
> How does /tg/ attract so many different opinions to the point of suspected sabotage/trolling.

image
Clearly just here to sabotage right? Go shitpost some other PR.

**Comment by Shadowflame909** [comment]
Can't blame a player for being suspect.

Your opinions on these features are really something different.

No one wants to buy something for the price of an e-sword and start dealing with the effects of what is essentially a poison.

**Comment by kriskog** [comment]
And when it's in we can evaluate if people are buying it or not. If they are not, the price can be adjusted accordingly. Until then, I don't want to open up for comboing adrenals differently than we currently can. The main reason for that is they are better than than some of you seem to fear, just different than before.

**Comment by ExcessiveUseOfCobblestone** [comment]
People not buying it is an intended aspect of the PR though because it's not meant to be a MUST HAVE.

**Comment by Shadowflame909** [comment]
I get that. But I think this PR has 3 core issues that need addressing for the best Quality of Life.

1. Turning this "Combat-Implant" into an "Escape-Implant" is very risky when you make one of the downsides a stackable damage source:

For example, let's say you're a Nukie on the run from a robust sec officer. They hit you with a stun-baton and you go down. They keep harm-batoning you, stacking on brute damage. When you try to escape with the implant the 15 temporary oxygen + whatever else puts you in critical condition.

Because of how oxygen damage works, even if they don't kill you. The o2 loss constantly rising will do you in anyways. High enough toxin also rises by itself.

I think you need another non-stackable damage-type if you still want that as a downside. Like I said maybe delayed stamina, or minor Brain Damage, also minor organ damage if that doesn't stack. (Unsure but I think it should.)  

But yeah the point on that is trying to escape at a low enough health-point and it quickly turns into accidental suicide. Thanks, damage code leading the main damage sources to bottom-up at 100%.

2. This PR doesn't address if its a must-have or not. Adrenals is a counter-play and a reaction like you said cobblestone. No matter how much it gets nerfed. People still will try to use a counter-play. So I think it'll still be a must-have for certain antags. IE lings and Nukies. It's just the general Quality Of Life for the players who use it. Well, they won't be so happy, as it'd be more irritating.

3. The main weariness and uncertainty this pull request causes are necessary reactions because antag changes will be blocked for a month. So if this PR is bad and greatly reduces the survivability of an antag and the usefulness of adrenals for escaping. Antag players are just gonna have to deal with it for a total month.

So you gotta understand it's pretty much a one-way ticket for perfection.

**Comment by ExcessiveUseOfCobblestone** [comment]
The alternative is they get removed wholesale.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Turning this "Combat-Implant" into an "Escape-Implant" is very risky when you make one of the downsides a stackable damage source:

You should play with that in mind. If you get stunned enough to rack up that kind of damage then sorry you lose.

> This PR doesn't address if its a must-have or not. Adrenals is a counter-play and a reaction like you said cobblestone. No matter how much it gets nerfed. People still will try to use a counter-play. So I think it'll still be a must-have for certain antags. IE lings and Nukies. It's just the general Quality Of Life for the players who use it. Well, they won't be so happy, as it'd be more irritating.

You can use it to counterplay still, but it comes with downsides. Is it REALLY worth it? Maybe. That's the goal of this PR, to drop down its effectiveness so it's still usable but not necessarily a must-have.

> The main weariness and uncertainty this pull request causes are necessary reactions because antag changes will be blocked for a month. So if this PR is bad and greatly reduces the survivability of an antag and the usefulness of adrenals for escaping. Antag players are just gonna have to deal with it for a total month.

You don't have to use the item if you don't think you'll find yourself in a situation where the cost of it is worth the benefit. that isthe point of the PR.

**Comment by Shadowflame909** [comment]
So you're telling me an 8tc item that has the intended purpose of getting you out of a close call "I lose" situation should fuck you over if the "I lose" situation has you taking some serious damage.

Not everyone is lexia black levels of robust cobblestone.

That's very much a noobbait death-trap.

I think the alternative is much better if the only option is ruining the rounds of players with a placebo that turns out to be full of poison.

**Comment by ExcessiveUseOfCobblestone** [comment]
You're suppose to use it before so you DON'T get blasted by the shotgun, not tank the shotgun. You don't have to be lexia black to not be braindead in seeing how this is suppose to function.

What does TC have to do with the power level of the item relative to the core game? It's monopoly money used to protect players from an unbeatable combo of gear.

**Comment by Shadowflame909** [comment]
If you knrwy you we're going to get shot at by anything. You  wouldn't need the implant.

It's the counter. The insurance, the safeguard. And when my trap card nog only doesn't work, but destroys me instead. Something's wrong here!

So forget migating immobilities if it's just going to be a lie that only see's use in situations it's not necessary in. And completely screws you in situations it is!

IE: Being harmbatonned or Beanbag slugged to death.

If the desire is to escape situations where death is not imminent. Cuffbreaker implants exist.

Don't play the playerbase with an item that costs as much as that universal doorbreaking enslavement tool. Or floor tile disguiser, or camo box, also that laser sword. 

Except this item will kill you when you need it most!

**Comment by ExcessiveUseOfCobblestone** [comment]
Getting shot once with anything wouldn't result in going down. If someone gets the jump one you AND manages to shoot you times take the L.

You can also tie these with syndismokes for self-serve healing.

**Comment by Shadowflame909** [comment]
beanbag slugs and rubber shot will do you in quite rapidly. But that's not even the situation I'm talking about.

What is the intended use for this item anymore stone of cobble? If it's to escape situations you don't see yourself dying in. There's always cuffbreaker implants.

You get hit too many times and you add a whole 22.5+ damage thanks to this implant after one use.  It's going to screw you over.

My traitor bucks are pretty valuable in getting me some items to have a good offense with. if I have to buy syndi-smokes on top of adrenals to get some intended functionality out of this. Then that's not very cool.

I shouldn't need to speed off to medbay, hoping the chemist quickly did their job and smart sec-officers won't realize where I have to go if I don't want this damage to stick on me. 

You don't have any issue with pricing whatsoever because TC is just meme cash to you. So i'm uncertain where you'd draw the line. Would it get merged if it was a 20 tc item in its current state?

If not, then TC does have some inherent worth on how good an item should be? Doesn't it?

Lastly, since I'm just getting into another endless argument with you:

> If the desire is to escape situations where death is not imminent. Cuffbreaker implants exist.

Don't play the playerbase with an item that costs as much as that universal doorbreaking enslavement tool. Or floor tile disguiser, or camo box, also that laser sword.

**Comment by ExcessiveUseOfCobblestone** [comment]
cuffbreakers imply you are going to be cuffed instead of killed, which is a pretty big implication.

TC should not dictate power, power should dictate TC since that's what the mechanic is for. If it turns out that it's not as warranting of the price then it can be lowered. It shouldn't be kept at a certain power level because it currently cost X amount of monopoly money.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -1234,3 +1250,39 @@
...
+	..()
+	if(prob(30) && iscarbon(M))
+		var/obj/item/I = M.get_active_held_item()
+		if(I && M.dropItemToGround(I))
+			to_chat(M, "<span class='notice'>Your hands spaz out and you drop what you were holding!</span>")
+	if(prob(40))
+		M.losebreath++
+		M.adjustOxyLoss(1, 0)
+	M.adjustStaminaLoss(-10, 0)
+	M.adjustToxLoss(0.5, 0)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I do agree this is a bit much though since

7.5 damage you can't heal out of unless you see medbay (or use puke juice which comes with side effects) on top of >15 oxy damage is very steep.

Comment (kriskog) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I get where you are coming from however this is in line with how the better chems affect you, in this case the oxy (toned down the prob) and tox (left as is) damage is from aranesp. Ling adrenals also give you tox, so there is some precedent. If you think it's a deal breaker, we can adjust it.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: ah I see, it's just a lesser form of aran?

Comment (kriskog) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: The stamina and oxy/tox effects are, yeah. As I said, the tox damage is left as is, since it's pretty low to begin with. 

For reference, aranesp has:
M.adjustStaminaLoss(-18, 0)
M.adjustToxLoss(0.5, 0)
if(prob(50))
	M.losebreath++
	M.adjustOxyLoss(1, 0)

**Comment by Shadowflame909** [comment]
Ling adrenals are completely infinite gamer. while this can only be used twice.

You could also try flipping the ling adrenals and traitor adrenals honestly.

Lings are already immortal murderboning machines, that won't be stopped by any medical changes whatsoever. While tator tots most definitively will.

Why are you giving the buffed up adrenals version to the infinite revivable antag with a hideable e-sword and the ability to constantly change disguises with a tap of a button.

Whilst the antag with only 2 buttons has to deal with all of these downsides

Very silly!

**Comment by AdamElTablawy** [comment]
the alternative to merging a PR nobody likes because it poisons you is removing adrenals wholesale

it's already limited in use and expensive, it doesn't need to injure you considering it no longer instantly gets you up from stuns. the poisoning is entirely unnecessary.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm not going to merge the ling changes since the other PR was merged, and honestly would like to just have the dropping hand mechanic instead of C2ing it.

**Comment by ExcessiveUseOfCobblestone** [comment]
@actioninja I told him i'd be okay with the ling changes if you are (since you merged that other PR)

**Comment by actioninja** [comment]
I think adrenals are fundamentally a stupid item that are indicative of a broken combat system but this actually might be an overnerf. Oh well, hit it.

**Comment by Tumesuo** [comment]
Not good 0/5 Screenshakes make it hard to escape and see what the fuck is happening/where you are running, dropping items while running might as well be a fucking death if you drop a gun on the first tick you are escaping and they gun you down with ya own gun. I am salty because I died from dropping my gun and them batoning me to death because of this. It would be better to have a mouth pill with Meth, Psicodine and Mannitol instead of spending 8 tc on this. Adrenals never again. (™)

**Comment by kriskog** [comment]
Yes, and do enjoy your meth brain traumas. If you used the adrenals and tried to keep fighting, that stuff will happen. How they got close to you while you ignore slowdowns and have a speed buff is beyond me.

**Comment by Tumesuo** [comment]
> Yes, and do enjoy your meth brain traumas. 
If you add mannitol to meth pills under 9 units and maybe some pump up instead of psicodine. You will not get traumas and it will be better than adrenals??
